cornice/schemas.py

```diff
@@ -175,3 +175,78 @@ def _validate_fields(location, data):
         for param in set(params) - set([attr.name for attr in attrs]):
             request.errors.add('body' if param in body else 'querystring',
                                param, msg % param)
+
+
+def validate_matchdict(matchdict_types):
```

Comment:
I believe that it could be part of Cornice. I find it much more elegant to use a validator like this than [what we did in cliquet](https://github.com/mozilla-services/cliquet/blob/master/cliquet/resource.py#L233) for example.


Even if I don't like it too much, Cornice provides a way to validate several parts of the request using the ``location`` argument in a schema. Shouldn't we take advantage of it instead ?

```python
class Schema():
    id = colander.SchemaNode(colander.Int, location='path')
    phone = colander.SchemaNode(colander.String, location='body')
```